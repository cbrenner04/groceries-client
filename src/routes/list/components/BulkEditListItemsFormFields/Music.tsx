import React, { type ChangeEventHandler } from 'react';

import { CheckboxField, TextField } from 'components/FormFields';

export interface IMusicProps {
  artist: string;
  clearArtist: boolean;
  handleClearArtist: ChangeEventHandler;
  album: string;
  clearAlbum: boolean;
  handleClearAlbum: ChangeEventHandler;
  handleInput: ChangeEventHandler;
}

const Music: React.FC<IMusicProps> = (props): React.JSX.Element => (
  <React.Fragment>
    <TextField
      name="artist"
      label="Artist"
      value={props.artist}
      handleChange={props.handleInput}
      placeholder="Sir Mix-a-Lot"
      disabled={props.clearArtist}
      child={
        <CheckboxField
          name="clearArtist"
          label="Clear artist"
          handleChange={props.handleClearArtist}
          value={props.clearArtist}
          classes="ms-1 mt-1"
        />
      }
    />
    <TextField
      name="album"
      label="Album"
      value={props.album}
      handleChange={props.handleInput}
      placeholder="Mack Daddy"
      disabled={props.clearAlbum}
      child={
        <CheckboxField
          name="clearAlbum"
          label="Clear album"
          handleChange={props.handleClearAlbum}
          value={props.clearAlbum}
          classes="ms-1 mt-1"
        />
      }
    />
  </React.Fragment>
);

export default Music;
