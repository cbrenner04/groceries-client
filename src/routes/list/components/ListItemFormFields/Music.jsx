import React from 'react';
import PropTypes from 'prop-types';

import { TextField, CategoryField, CheckboxField } from '../../../../components/FormFields';

const MusicFormFields = (props) => (
  <>
    <TextField
      name="title"
      label="Title"
      value={props.title}
      handleChange={props.inputChangeHandler}
      placeholder="Baby Got Back"
    />
    <TextField
      name="artist"
      label="Artist"
      value={props.artist}
      handleChange={props.inputChangeHandler}
      placeholder="Sir Mix-a-Lot"
    />
    <TextField
      name="album"
      label="Album"
      value={props.album}
      handleChange={props.inputChangeHandler}
      placeholder="Mack Daddy"
    />
    <CategoryField
      category={props.category || ''}
      categories={props.categories || []}
      handleInput={props.inputChangeHandler}
    />
    {props.editForm && (
      <CheckboxField
        name="purchased"
        label="Purchased"
        value={props.purchased || false}
        handleChange={props.inputChangeHandler}
        classes="mb-3"
      />
    )}
  </>
);

MusicFormFields.propTypes = {
  title: PropTypes.string.isRequired,
  artist: PropTypes.string.isRequired,
  album: PropTypes.string.isRequired,
  purchased: PropTypes.bool,
  editForm: PropTypes.bool,
  category: PropTypes.string,
  categories: PropTypes.arrayOf(PropTypes.string),
  inputChangeHandler: PropTypes.func.isRequired,
};

export default MusicFormFields;
