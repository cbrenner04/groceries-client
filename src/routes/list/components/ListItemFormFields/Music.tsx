import React, { ChangeEventHandler } from 'react';

import { TextField, CategoryField, CheckboxField } from '../../../../components/FormFields';

interface IMusicFormFieldsProps {
  title: string;
  artist: string;
  album: string;
  purchased?: boolean;
  editForm?: boolean;
  category?: string;
  categories?: string[];
  inputChangeHandler: ChangeEventHandler;
}

const MusicFormFields: React.FC<IMusicFormFieldsProps> = (props) => (
  <>
    <TextField
      name="title"
      label="Title"
      value={props.title}
      handleChange={props.inputChangeHandler}
      placeholder="Baby Got Back"
    />
    <TextField
      name="artist"
      label="Artist"
      value={props.artist}
      handleChange={props.inputChangeHandler}
      placeholder="Sir Mix-a-Lot"
    />
    <TextField
      name="album"
      label="Album"
      value={props.album}
      handleChange={props.inputChangeHandler}
      placeholder="Mack Daddy"
    />
    <CategoryField
      category={props.category ?? ''}
      categories={props.categories ?? []}
      handleInput={props.inputChangeHandler}
    />
    {props.editForm && (
      <CheckboxField
        name="purchased"
        label="Purchased"
        value={props.purchased ?? false}
        handleChange={props.inputChangeHandler}
        classes="mb-3"
      />
    )}
  </>
);

export default MusicFormFields;
