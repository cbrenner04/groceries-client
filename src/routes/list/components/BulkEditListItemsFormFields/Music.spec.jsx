import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import Music from './Music';

describe('Music', () => {
  const props = {
    artist: 'foo',
    clearArtist: false,
    handleClearArtist: jest.fn(),
    handleInput: jest.fn(),
    album: 'bar',
    clearAlbum: false,
    handleClearAlbum: jest.fn(),
  };

  it('renders artist input enabled when clearArtist is false', () => {
    props.clearArtist = false;
    const { container, getByLabelText } = render(<Music {...props} />);

    expect(container).toMatchSnapshot();
    expect(getByLabelText('Artist')).toBeEnabled();
  });

  it('renders artist input disabled when clearArtist is true', () => {
    props.clearArtist = true;
    const { container, getByLabelText } = render(<Music {...props} />);

    expect(container).toMatchSnapshot();
    expect(getByLabelText('Artist')).toBeDisabled();
  });

  it('renders album input enabled when clearAlbum is false', () => {
    props.clearAlbum = false;
    const { container, getByLabelText } = render(<Music {...props} />);

    expect(container).toMatchSnapshot();
    expect(getByLabelText('Album')).toBeEnabled();
  });

  it('renders album input disabled when clearAlbum is true', () => {
    props.clearAlbum = true;
    const { container, getByLabelText } = render(<Music {...props} />);

    expect(container).toMatchSnapshot();
    expect(getByLabelText('Album')).toBeDisabled();
  });

  it('handles change in input and checkbox', async () => {
    props.clearArtist = false;
    const { getByLabelText, getAllByRole } = render(<Music {...props} />);

    fireEvent.change(getByLabelText('Artist'), { target: { value: 'a' } });

    expect(props.handleInput).toHaveBeenCalled();

    fireEvent.click(getAllByRole('checkbox')[0]);

    expect(props.handleClearArtist).toHaveBeenCalled();

    fireEvent.change(getByLabelText('Album'), { target: { value: 'a' } });

    expect(props.handleInput).toHaveBeenCalled();

    fireEvent.click(getAllByRole('checkbox')[1]);

    expect(props.handleClearAlbum).toHaveBeenCalled();
  });
});
