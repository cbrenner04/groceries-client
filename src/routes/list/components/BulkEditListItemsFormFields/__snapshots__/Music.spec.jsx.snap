// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Music renders album input disabled when clearAlbum is true 1`] = `
<div>
  <div
    class="mb-3"
  >
    <label
      class="form-label"
      for="artist"
    >
      Artist
    </label>
    <input
      class="form-control"
      disabled=""
      id="artist"
      name="artist"
      placeholder="Sir Mix-a-Lot"
      type="text"
      value="foo"
    />
    <div
      class="ms-1 mt-1 mb-3"
    >
      <div
        class="form-check"
      >
        <input
          checked=""
          class="form-check-input"
          id="clearArtist"
          name="clearArtist"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="clearArtist"
          title=""
        >
          Clear artist
        </label>
      </div>
    </div>
  </div>
  <div
    class="mb-3"
  >
    <label
      class="form-label"
      for="album"
    >
      Album
    </label>
    <input
      class="form-control"
      disabled=""
      id="album"
      name="album"
      placeholder="Mack Daddy"
      type="text"
      value="bar"
    />
    <div
      class="ms-1 mt-1 mb-3"
    >
      <div
        class="form-check"
      >
        <input
          checked=""
          class="form-check-input"
          id="clearAlbum"
          name="clearAlbum"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="clearAlbum"
          title=""
        >
          Clear album
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`Music renders album input enabled when clearAlbum is false 1`] = `
<div>
  <div
    class="mb-3"
  >
    <label
      class="form-label"
      for="artist"
    >
      Artist
    </label>
    <input
      class="form-control"
      disabled=""
      id="artist"
      name="artist"
      placeholder="Sir Mix-a-Lot"
      type="text"
      value="foo"
    />
    <div
      class="ms-1 mt-1 mb-3"
    >
      <div
        class="form-check"
      >
        <input
          checked=""
          class="form-check-input"
          id="clearArtist"
          name="clearArtist"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="clearArtist"
          title=""
        >
          Clear artist
        </label>
      </div>
    </div>
  </div>
  <div
    class="mb-3"
  >
    <label
      class="form-label"
      for="album"
    >
      Album
    </label>
    <input
      class="form-control"
      id="album"
      name="album"
      placeholder="Mack Daddy"
      type="text"
      value="bar"
    />
    <div
      class="ms-1 mt-1 mb-3"
    >
      <div
        class="form-check"
      >
        <input
          class="form-check-input"
          id="clearAlbum"
          name="clearAlbum"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="clearAlbum"
          title=""
        >
          Clear album
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`Music renders artist input disabled when clearArtist is true 1`] = `
<div>
  <div
    class="mb-3"
  >
    <label
      class="form-label"
      for="artist"
    >
      Artist
    </label>
    <input
      class="form-control"
      disabled=""
      id="artist"
      name="artist"
      placeholder="Sir Mix-a-Lot"
      type="text"
      value="foo"
    />
    <div
      class="ms-1 mt-1 mb-3"
    >
      <div
        class="form-check"
      >
        <input
          checked=""
          class="form-check-input"
          id="clearArtist"
          name="clearArtist"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="clearArtist"
          title=""
        >
          Clear artist
        </label>
      </div>
    </div>
  </div>
  <div
    class="mb-3"
  >
    <label
      class="form-label"
      for="album"
    >
      Album
    </label>
    <input
      class="form-control"
      id="album"
      name="album"
      placeholder="Mack Daddy"
      type="text"
      value="bar"
    />
    <div
      class="ms-1 mt-1 mb-3"
    >
      <div
        class="form-check"
      >
        <input
          class="form-check-input"
          id="clearAlbum"
          name="clearAlbum"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="clearAlbum"
          title=""
        >
          Clear album
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`Music renders artist input enabled when clearArtist is false 1`] = `
<div>
  <div
    class="mb-3"
  >
    <label
      class="form-label"
      for="artist"
    >
      Artist
    </label>
    <input
      class="form-control"
      id="artist"
      name="artist"
      placeholder="Sir Mix-a-Lot"
      type="text"
      value="foo"
    />
    <div
      class="ms-1 mt-1 mb-3"
    >
      <div
        class="form-check"
      >
        <input
          class="form-check-input"
          id="clearArtist"
          name="clearArtist"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="clearArtist"
          title=""
        >
          Clear artist
        </label>
      </div>
    </div>
  </div>
  <div
    class="mb-3"
  >
    <label
      class="form-label"
      for="album"
    >
      Album
    </label>
    <input
      class="form-control"
      id="album"
      name="album"
      placeholder="Mack Daddy"
      type="text"
      value="bar"
    />
    <div
      class="ms-1 mt-1 mb-3"
    >
      <div
        class="form-check"
      >
        <input
          class="form-check-input"
          id="clearAlbum"
          name="clearAlbum"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="clearAlbum"
          title=""
        >
          Clear album
        </label>
      </div>
    </div>
  </div>
</div>
`;
